drafts/liberty-dev-mode-gradle.adoc

```diff
@@ -0,0 +1,128 @@
+---
+layout: post
+title: "Rapid application development with Gradle and Open Liberty's dev mode"
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/u/28316667
+author_github: https://github.com/rzgry
+seo-title: Rapid application development with Gradle and Open Liberty's dev mode - OpenLiberty.io
+seo-description: Develop Java applications quicker using Open Liberty development mode with Gradle.
+blog_description: "Develop Java applications quicker using Open Liberty development mode with Gradle."
+---
+= Open Liberty development mode for Gradle
+Ryan Zegray <https://github.com/rzgry>
+
+Open Liberty development mode, or dev mode, now works with Gradle projects. Open Liberty dev mode allows you to develop applications with any text editor or IDE by providing hot reload and deployment, on demand testing, and debugger support. Your code is automatically compiled and deployed to your running server, making it easy to iterate on your changes. You can run tests on demand or even automatically so that you can get immediate feedback on your changes. You can also attach a debugger at any time to debug your running application.
+
+In this post, we show how dev mode allows you to rapidly develop Java applications using Open Liberty dev mode and Gradle.  We demonstrate this with an example scenario of enabling a MicroProfile health check endpoint in a REST microservice.
+
+You can follow along using your own project or by cloning the https://github.com/OpenLiberty/demo-devmode[demo-devmode] sample project.
+
+----
+git clone https://github.com/OpenLiberty/demo-devmode.git
+----
+
+== Including the liberty gradle plugin
+
+For Gradle projects, dev mode is run through the Liberty Gradle Plugin version `3.0` or above. If you are using the sample project you will notice the project defines the plugin in the projects `build.gradle` file.
+
+[source,groovy]
+----
+buildscript {
+    repositories {
+        mavenCentral()
+    }
+    dependencies {
+        classpath 'io.openliberty.tools:liberty-gradle-plugin:3.0'
+    }
+}
+----
+
+== Starting the application using dev mode
+
+Start dev mode by running the following command in your project: `gradle libertyDev`. This will start the server in dev mode along with installing features and deploying your application.
+
+== Automatically recompiling and deploying to the running server
+
+Once dev mode is started, source code changes are automatically compiled and deployed to the running server. If any features are added to the `server.xml` file, they are installed automatically.
+
+For example, adding the the `mpHealth-2.1` feature to server.xml to enable health checks will make the health endpoint accessible at http://localhost:9080/health.
+
+
+image::/img/blog/dev_mode_server_xml.png[mp-health Server XML]
+image::/img/blog/dev_mode_health_check_1.png[MicroProfile Health check]
```

Comment:
Resize these images since they are too large.  You could resize by adding something like `, width=400]` at the end of the image tag.